src/ParallelAlgorithms/Interpolation/Targets/LineSegment.hpp

```diff
@@ -119,26 +119,31 @@ namespace TargetPoints {
 /// \brief Computes points on a line segment.
 ///
 /// For requirements on InterpolationTargetTag, see InterpolationTarget
-template <typename InterpolationTargetTag, size_t VolumeDim>
+///
+/// \note This takes a `tmpl::size_t<Dim>` as its second template argument so
+/// that it is consistent with the other Targets having only type template
+/// parameters and no non-type template parameters.
+template <typename InterpolationTargetTag, typename VolumeDim>
```

Comment:
I see why you did this, but I think we have to find another solution.. It seems impractical to limit template parameters to `typename` just so they're easier to handle in protocols. Brigand does this to an extent, and `::Tags::deriv` does this for compatibility with `db::wrap_tags_in`, but I've found it very annoying and meant to refactor that for a while now. I think it would be much too limiting to impose a rule like "if a class conforms to a protocol, it can only have `typename` template parameters". We should instead improve our tools to handle this situation.

Reply:
Yeah I wasn't a fan of doing this but it was necessary to get everything to compile.

It's not that protocols only require `typename` templates, it's just that when doing `template template` expressions, all template parameters have to be similar or you have to know you're always getting a `non-type` parameter so you can account for it. It's not specific to protocols.